libs/logging/src/logger.test.ts

```diff
@@ -101,3 +111,267 @@ test('logging from a client side app without sending window.kiosk does NOT log t
   await logger.log(LogEventId.AdminCardInserted, 'admin');
   expect(console.log).not.toHaveBeenCalled();
 });
+
+test('verify that client side apps are configured properly', () => {
+  for (const source of CLIENT_SIDE_LOG_SOURCES) {
+    expect(source in DEVICE_TYPES_FOR_APP).toBeTruthy();
+  }
+});
+
+describe('test cdf conversion', () => {
+  test('builds device and election info properly', () => {
+    const logger = new Logger(LogSource.VxAdminApp);
+    const cdfLogContent = logger.buildCDFLog(
+      electionMinimalExhaustiveSampleDefintion,
+      '',
+      '12machine34',
+      'thisisacodeversion',
+      'admin'
+    );
+    const cdfLog = JSON.parse(cdfLogContent);
```

Comment:
Using `JSON.parse` here means TS treats `cdfLog` and everything derived from it as `any`, which means refactoring on the type this is supposed to have does not work. For example, I tried to rename `CdfDevice#id` to `CdfDevice#Id` and line 141 was not updated because TS doesn't know that `id` is the same as this one.

Reply:
do you think I should use zod here then? That felt like overkill to me just for the tests when we don't need to parse in the actual code but I don't feel strongly.

Reply:
Why don't we need to parse it in the actual code?

Reply:
Well I guess if we end up needing to import logs then we will (this is in the VVSG spec but Monica seems to think that it doesn't make sense that we would need to do that and told me to ignore it for now), but otherwise we are never reading a CDF log file in the actual code, just converting to it as an output.